src/commands/set.js

```diff
@@ -41,24 +43,34 @@ const writeConfigToFile = newConfig => {
 
 const checkBoolean = value => ['true', 'false'].includes(value);
 
-const setNestedConfigProperty = newValue => (obj, pathComponent, i, path) => {
-	if (i === path.length - 1) {
+const setNestedConfigProperty = newValue => (
+	obj,
+	pathComponent,
+	i,
+	dotNotationArray,
+) => {
+	if (i === dotNotationArray.length - 1) {
+		if (obj[pathComponent] === undefined) {
```

Comment:
This should be `if(obj === undefined)` since we are inserting the value below.
I think newly created value should be insertable.
ex)
if key `api` doesn't exist, throw error, but if key `api` exists but `port` doesn't, then insert `port`